### PR TITLE
sshfs: use modern meson build syntax and undeprecate

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

closes #162353

### CODE REVIEW

File: Formula/s/sshfs.rb [hunk 1]
```diff
@@ -9,8 +9,6 @@ class Sshfs < Formula
     sha256 x86_64_linux: "a98d273e64706971684935a3ae87da16b1dda98f7289eb79e82f4cdfb7f12bb8"
   end
 
-  deprecate! date: "2023-01-01", because: :repo_archived
-
```
Comment (SMillerDev) [inline Formula/s/sshfs.rb:13]: Does this no longer apply? https://github.com/libfuse/sshfs?tab=readme-ov-file#development-status

**Review by SMillerDev — CHANGES_REQUESTED** [review]

File: Formula/s/sshfs.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/s/sshfs.rb:13]: sounds fair, but the repo is no longer archived? Let me ask upstream to check with maintenance status.

**Comment by chenrui333** [comment]
cc @h4sh5 @Nikratio for thoughts?

**Comment by h4sh5** [comment]
yes, SSHFS is active and not unmaintained as also pointed out here https://github.com/libfuse/sshfs/issues/292#issuecomment-1890381986

**Comment by SMillerDev** [comment]
That comment says that the README is accurate, and that README says that it's essentially in maintenance-only mode. I'd say it needs to stay deprecated

**Comment by anatol** [comment]
Is there any chance to move with this change forward?

As it was said previoously `sshfs` project is not archived anymore. And what is more important it is a tool that is actively used, and there simply no other alternatives to sshfs approach at MacOS.

**Comment by p-linnane** [comment]
Given that @h4sh5 is a contributor with write access to that project, and they have said it's "active and not unmaintained", I am fine to undeprecate this.

**Review by p-linnane — APPROVED** [review]

**Review by SMillerDev — APPROVED** [review]

**Pull request merged**